plugins/modules/dcnm_policy.py

```diff
@@ -0,0 +1,1072 @@
+
+    def dcnm_translate_switch_info(self, config, ip_sn, hn_sn):
+
+       if (None is config):
+         return
+
+       for cfg in config:
+
+           index = 0
+
+           if (None is cfg.get('switch', None)):
+               continue
+           for sw_elem in cfg['switch']:
+               addr_info = dcnm_get_ip_addr_info (self.module, sw_elem['ip'], ip_sn, hn_sn)
+               cfg['switch'][index]['ip'] = addr_info
+               index = index + 1
+
+    def dcnm_translate_config (self, config):
+
+        # In the playbook, switches is given as a separate object at policy configuration level. 
+        # We will remove it from there and add it to individual policies
+
+        # Get the position of the matching dict
+        pos = next((index for (index, d) in enumerate(config) if "switch" in d), None)
+
+        if (pos == None):
+            return config
+
+        sw_dict = config.pop(pos)
+
+        # 'switches' contaions the switches related configuration items from playbook. Process these
+        # and add the same to individual policy items as appropriate
+
+        new_config = []
+        for sw in sw_dict['switch']:
+
+            # Check if policies are included in the switch config. If so these policies will override
+            # all other polcy configs given in the playbook globally. If not the switch can be included 
+            # in all individaul policy items
+
+            if (sw.get ('policies', None) != None):
+
+                # 'policies are specified at switch level. Add eachj of these policies to 'config' object 
+                # along with the switch information
+
+                for pol in sw['policies']:
+
+                    if (pol.get('switch', None) == None):
+                        pol['switch'] = []
+                    if (sw['ip'] not in pol['switch']):
+                        pol['switch'].append(sw['ip'])
+                    #if (pol not in new_config):
+                    new_config.append(pol)
+            else:
+
+                # This switch does not have any policies included. Add this switch to all policies in the 
+                # playbook config
+
+                for cfg in config:
+
+                    if (cfg.get('switch', None) == None):
+                        cfg['switch'] = []
+                    if (sw['ip'] not in cfg['switch']):
+                        cfg['switch'].append(sw['ip'])
+        if (new_config != []):                 
+            config.extend(new_config)
+
+        return config
+
+def main():
+
+    """ main entry point for module execution
+    """
+    element_spec = dict(
+        fabric=dict(required=True, type='str'),
+        config=dict(required=False, type='list'),
+        state=dict(type='str', default='merged',
+                   choices = ['merged', 'deleted', 'query']),
+        deploy=dict(required=False, type='bool', default=True)
+    )
+
+    with open ("check.log", "w+") as f:
```

Comment:
This write to file should be removed for the final version right?

Reply:
Removed